lib/hdb/hdb.h

```diff
@@ -294,7 +294,7 @@ typedef struct HDB {
     /**
      * Check if s4u2self is allowed from this client to this server
      */
-    krb5_error_code (*hdb_check_s4u2self)(krb5_context, struct HDB *, hdb_entry_ex *, krb5_const_principal);
+    krb5_error_code (*hdb_check_s4u2self)(krb5_context, struct HDB *, hdb_entry_ex *, hdb_entry_ex *);
```

Comment:
Check 5aaf12351aa30ae2e9c657080b8566397c0da6bd hasn't made its way into any releases, otherwise you'll also need to bump the HDB interface version.